core/src/main/java/org/radargun/traits/Queryable.java

```diff
@@ -27,27 +27,52 @@
 
    /**
     * The instance should be reusable, but not thread-safe.
+    * Conditions defined after groupBy call are meant to be part of the HAVING clause.
     */
    interface QueryBuilder {
       QueryBuilder subquery();
-      QueryBuilder eq(String attribute, Object value);
-      QueryBuilder lt(String attribute, Object value);
-      QueryBuilder le(String attribute, Object value);
-      QueryBuilder gt(String attribute, Object value);
-      QueryBuilder ge(String attribute, Object value);
-      QueryBuilder between(String attribute, Object lowerBound, boolean lowerInclusive, Object upperBound, boolean upperInclusive);
-      QueryBuilder isNull(String attribute);
-      QueryBuilder like(String attribute, String pattern);
-      QueryBuilder contains(String attribute, Object value);
+      QueryBuilder eq(SelectExpression selectExpression, Object value);
+      QueryBuilder lt(SelectExpression selectExpression, Object value);
+      QueryBuilder le(SelectExpression selectExpression, Object value);
+      QueryBuilder gt(SelectExpression selectExpression, Object value);
+      QueryBuilder ge(SelectExpression selectExpression, Object value);
+      QueryBuilder between(SelectExpression selectExpression, Object lowerBound, boolean lowerInclusive, Object upperBound, boolean upperInclusive);
+      QueryBuilder isNull(SelectExpression selectExpression);
+      QueryBuilder like(SelectExpression selectExpression, String pattern);
+      QueryBuilder contains(SelectExpression selectExpression, Object value);
       QueryBuilder not(QueryBuilder subquery);
       QueryBuilder any(QueryBuilder... subqueries);
-      QueryBuilder orderBy(String attribute, SortOrder order);
-      QueryBuilder projection(String... attribute);
+      QueryBuilder orderBy(SelectExpression selectExpression, SortOrder order);
```

Comment:
Could you issue a JIRA for that to Infinispan?
Anyway, I'd like to see a clean semanthics in the RadarGun API, and I don't know what this is supposed to do. So, either comment and define it, or limit it to arguments with obvious meaning (I know that there are not comments on the other methods in this class, but these are IMO quite obvious).